test/hook/context/generator.go

```diff
@@ -109,12 +113,13 @@ func (b *BindingContextController) Run() (string, error) {
 	}
 
 	b.HookCtrl = controller.NewHookController()
-	b.HookCtrl.InitKubernetesBindings(testHook.GetConfig().OnKubernetesEvents, b.Manager)
+	b.HookCtrl.InitKubernetesBindings(testHook.GetConfig().OnKubernetesEvents, b.BindingManager)
+	b.HookCtrl.InitScheduleBindings(testHook.GetConfig().Schedules, b.ScheduleManager)
+	b.HookCtrl.EnableScheduleBindings()
```

Comment:
This call to EnableScheduleBindings() is redundant. There is no need of fully initialized ScheduleManager object because BindingContext is generated on demand by RunSchedule and not by channel monitoring.

Back then ScheduleManager was used to return error on initialization if `crontab` was not valid. Now, this check is performed when the configuration is parsed:

```
110: testHook, err = testHook.WithConfig([]byte(b.HookConfig))
```

I see no reason to start full-fledged ScheduleManager, it can be completely removed.

`nil` is good for previous line:
```
b.HookCtrl.InitScheduleBindings(testHook.GetConfig().Schedules, nil)
```

Reply:
Wrong: b.HookCtrl.EnableScheduleBindings() is necessary.